### PR TITLE
Migrate to Spark 2.4.0

### PR DESCRIPTION
**Pull request by pstutz**
Co-authored-by: Max Kießling <max.kiessling@neotechnology.com>

### CODE REVIEW

File: spark-cypher-testing/src/test/scala/org/opencypher/spark/SparkTests.scala [hunk 1]
```diff
@@ -32,8 +32,8 @@ import org.opencypher.spark.testing.CAPSTestSuite
 
 class SparkTests extends CAPSTestSuite {
 
-  // Example for: https://issues.apache.org/jira/browse/SPARK-23855
-  ignore("should correctly perform a join after a cross") {
+  // Reproduces https://issues.apache.org/jira/browse/SPARK-23855, which was relevant to Spark 2.2
+  it("should correctly perform a join after a cross") {
```
Comment (s1ck) [inline spark-cypher-testing/src/test/scala/org/opencypher/spark/SparkTests.scala]: The test could be removed now that this is fixed in Spark.

File: spark-cypher/src/main/scala/org/opencypher/spark/impl/SparkSQLExprMapper.scala [hunk 2]
```diff
@@ -382,7 +382,7 @@ object SparkSQLExprMapper {
 
         case Range(from, to, maybeStep) =>
           val stepCol = maybeStep.map(_.asSparkSQLExpr).getOrElse(ONE_LIT)
-          rangeUdf(from.asSparkSQLExpr, to.asSparkSQLExpr, stepCol)
+          functions.sequence(from.asSparkSQLExpr, to.asSparkSQLExpr, stepCol)
```
Comment (s1ck) [inline spark-cypher/src/main/scala/org/opencypher/spark/impl/SparkSQLExprMapper.scala]: nice

File: spark-cypher/src/main/scala/org/opencypher/spark/impl/table/SparkTable.scala [hunk 3]
```diff
@@ -229,12 +229,7 @@ object SparkTable {
           df.crossJoin(other.df)
 
         case _ =>
-
-          // TODO: the join produced corrupt data when the previous operator was a cross. We work around that by using a
-          // subsequent select. This can be removed, once https://issues.apache.org/jira/browse/SPARK-23855 is solved or we
-          // upgrade to Spark 2.3.0
-          val potentiallyCorruptedResult = df.safeJoin(other.df, joinCols, joinTypeString)
-          potentiallyCorruptedResult.select("*")
+          df.safeJoin(other.df, joinCols, joinTypeString)
```
Comment (s1ck) [inline spark-cypher/src/main/scala/org/opencypher/spark/impl/table/SparkTable.scala]: also nice!

File: spark-cypher-testing/src/test/scala/org/opencypher/spark/SparkTests.scala (hunk 1 above)
Comment (pstutz) [reply spark-cypher-testing/src/test/scala/org/opencypher/spark/SparkTests.scala]: We thought it was nice to keep it as an enabled test. Might be handy if there is ever a question about how to reproduce the Spark issue or if we back-port to an earlier Spark version.

**Comment by DarthMax** [comment]
Should be merged after the next release

**Comment by pstutz** [comment]
Release for `spark-cypher-0.2.7` is on Maven Central.

**Pull request merged**